src/k8s-engine/__tests__/k3sHelper.spec.ts

```diff
@@ -265,4 +266,129 @@ describe(K3sHelper, () => {
       await pendingInit;
     });
   });
+
+  describe('substituting versions', () => {
+    let subject: K3sHelper;
+    let config: settings.Settings;
+
+    beforeAll(() => {
+      config = settings.defaultSettings;
+      subject = new K3sHelper('x86_64');
+      subject['pendingInitialize'] = Promise.resolve();
+    });
+    const convertToSemvers = function(a: [string, string, string]) : [semver.SemVer, semver.SemVer, semver.SemVer] {
+      return [new semver.SemVer(a[0]), new semver.SemVer(a[1]), new semver.SemVer(a[2])];
+    };
+
+    describe('pickClosestVersion', () => {
+      test('picks same major version before pivot', () => {
```

Comment:
This chunk has a _lot_ of code repeats; it may be better to rewrite this as a [table-driven test](https://jestjs.io/docs/api#testeachtablename-fn-timeout) instead. (The test name may be one of the parameters.)

Reply:
I finally got both test blocks into tables, but it should be easier to do this. The docs I looked at said I could have an array of hashes, so I could use named parts, but in practice that's impossible.